lib/web/device_trust_test.go

```diff
@@ -36,85 +36,105 @@ import (
+		},
+		{
+			name:               "with empty path redirect_uri",
+			redirectURI:        "https://example.com",
+			expectedRedirectTo: "/web",
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+			fakeDevices := &fakeDevicesClient{}
+			wPack := newWebPack(
+				t,
+				1, /* numProxies */
+				withDevicesClientOverride(fakeDevices),
+			)
+			proxy := wPack.proxies[0]
+			aPack := proxy.authPack(t, "llama", nil /* roles */)
+			webClient := aPack.clt
+			ctx := context.Background()
+
+			query := make(url.Values)
+			query.Set("id", "my-token-id")
+			query.Set("token", "my-token-token")
+			if test.redirectURI != "" {
+				query.Set("redirect_uri", test.redirectURI)
 			}
 
-			redirected = true
-			if !assert.Len(t, via, 1, "CheckRedirect param via has an unexpected length") {
+			var redirected bool
+			var actualRedirectTo string
+			httpClient := webClient.HTTPClient()
+			httpClient.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+				if !redirected {
+					redirected = true
+					actualRedirectTo = req.URL.Path
+				}
 				return nil
 			}
-			src := via[0]
-
-			// Host didn't change, ie redirect is within the same Proxy.
-			assert.Equal(t, src.URL.Host, req.URL.Host, "CheckRedirect Host mismatch")
-			// Redirect target is as expected.
-			assert.Equal(t, "/web", req.URL.Path, "CheckRedirect dest Path mismatch")
-			// Redirect source is as expected.
-			assert.Regexp(t, "/webapi/devices/webconfirm$", src.URL.Path, "CheckRedirect src Path mismatch")
-
-			return nil
-		}
-
-		req, err := http.NewRequestWithContext(ctx, "GET", webClient.Endpoint("webapi", "devices", "webconfirm"), nil /* body */)
-		require.NoError(t, err, "NewRequestWithContext failed")
-		req.URL.RawQuery = query.Encode()
-
-		// Request using the httpClient, this shows we don't need the bearer token
-		// logic from webclient.
-		resp, err := httpClient.Do(req)
-		require.NoError(t, err, "GET /webapi/devices/webconfirm failed")
-		// Always drain and close the body.
-		io.Copy(io.Discard, resp.Body)
-		resp.Body.Close()
-
-		// Verify redirect and response status.
-		assert.True(t, redirected, "GET /webapi/devices/webconfirm didn't cause a redirect")
-		assert.Equal(t, 200, resp.StatusCode, "GET /webapi/devices/webconfirm code mismatch")
-
-		// Verify RPC call.
-		got := fakeDevices.resetConfirmRequests()
-		want := []*devicepb.ConfirmDeviceWebAuthenticationRequest{
-			{
-				ConfirmationToken: &devicepb.DeviceConfirmationToken{
-					Id:    "my-token-id",
-					Token: "my-token-token",
+
+			req, err := http.NewRequestWithContext(ctx, "GET", webClient.Endpoint("webapi", "devices", "webconfirm"), nil)
+			require.NoError(t, err, "NewRequestWithContext failed")
+			req.URL.RawQuery = query.Encode()
+
+			resp, err := httpClient.Do(req)
+			require.NoError(t, err, "GET /webapi/devices/webconfirm failed")
+			defer resp.Body.Close()
+			io.Copy(io.Discard, resp.Body)
```

Comment:
We can straight up close the body here, it looks like.

```suggestion
			io.Copy(io.Discard, resp.Body)
			resp.Body.Close()
```